### PR TITLE
[.NET] Fix unhandled GD0303 error for nested generic attribute types

### LINKED ISSUE #96585: GD0303 Error from MustBeVariantAnalyzer
### Tested versions

- Reproducible in: v4.3.stable.mono.official [77dcf97d8]

### System information

Godot v4.3.stable.mono - Windows 10.0.22631 - GLES3 (Compatibility) - NVIDIA RTX 2000 Ada Generation Laptop GPU (NVIDIA; 31.0.15.3847) - 13th Gen Intel(R) Core(TM) i7-13700H (20 Threads)

### Issue description

This C# script causes `MustBeVariantAnalyzer` to emit a `DG0303` error.

```cs
// Foo.cs
using System;
using Godot;

public partial class Foo : Node
{
    [A.B<int>]
    private int _bar;
}

public class A
{
    public class B<T> : Attribute { }
}
```

Minimal `.csproj`:
```xml
<Project Sdk="Godot.NET.Sdk/4.3.0">
	<PropertyGroup>
		<TargetFramework>net6.0</TargetFramework>
		<TargetFramework Condition=" '$(GodotTargetPlatform)' == 'android' ">net7.0</TargetFramework>
		<TargetFramework Condition=" '$(GodotTargetPlatform)' == 'ios' ">net8.0</TargetFramework>
		<EnableDynamicLoading>true</EnableDynamicLoading>
		<LangVersion>11</LangVersion>
	</PropertyGroup>
</Project>
```

My ultimate goal is to be able to use [Emik.SourceGenerators.Choices](https://www.nuget.org/packages/Emik.SourceGenerators.Choices), which can generate attributes such as this (with arbitrary nesting):
```cs
[Choice.A<int>.B<char>.C<string>.D<Node2D>.E<Node>]
// Choice is a nested class within Foo
public partial class Foo;
```

### Steps to reproduce

1. Create a new project with the above script and project file.
2. Attempt to build the project.

### Minimal reproduction project (MRP)

[Project.zip](https://github.com/user-attachments/files/16882680/Project.zip)

### PR DESCRIPTION
**Pull request by zaevi**
- Fixes #96585

### CODE REVIEW

**Review by raulsntos — COMMENTED** [review]
Thanks, it looks good. Can you add the fixed case to [`MustBeVariant.GD0301.cs`](https://github.com/godotengine/godot/blob/68f053bf82ff811c7a6f870832eb4bc4d09a62e5/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/TestData/Sources/MustBeVariant.GD0301.cs)? That way we verify that the PR fixes it and that we don't re-introduce the bug in the future.

File: modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/TestData/Sources/MustBeVariant.GD0301.cs [hunk 1]
```diff
@@ -395,6 +395,11 @@ public void MethodWithAttributeRidArray()
     public void MethodWithWrongAttribute()
     {
     }
+
+    [NestedGenericTypeAttributeContainer.NestedGenericTypeAttribute<bool>()]
```
Comment (zaevi) [inline modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/TestData/Sources/MustBeVariant.GD0301.cs:399]: Not sure if one case is enough

Comment (raulsntos) [reply modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators.Tests/TestData/Sources/MustBeVariant.GD0301.cs:399]: I think it should be enough, yeah.

**Review by raulsntos — APPROVED** [review]
LGTM. Thanks!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**